functions/src/business-logics/reply-logics.ts

```diff
@@ -8,3 +8,83 @@
 // TODO write onReplyDeleteLogic
 //  decrement comment repliesCount
 //  delete the notification for comment's author
+
+import {LogicConfig, LogicResultDoc} from "emberflow/lib/types";
+import {Entity} from "../db-structure";
+import {db} from "emberflow/lib";
+
+export const onReplyDeleteLogic: LogicConfig = {
+  name: "onReplyDeleteLogic",
+  actionTypes: ["delete"],
+  modifiedFields: "all",
+  entities: [Entity.Reply],
+  logicFn: async (action) => {
+    const {
+      eventContext: {docPath},
+      modifiedFields,
+    } = action;
+
+    const replyDoc = {
+      ...modifiedFields,
+    };
+
+    const replyRef = db.doc(docPath);
+    const commentDocRef = replyRef.parent.parent;
+
+    if (!commentDocRef) {
+      return {
+        name: "onReplyDeleteLogic",
+        status: "error",
+        documents: [],
+        message: `Invalid docPath at ${docPath}`,
+      };
+    }
+
+    const commentDocSnapshot = await commentDocRef.get();
+    const commentDoc = commentDocSnapshot.data();
+
+    if (commentDoc === undefined) {
+      return {
+        name: "onReplyDeleteLogic",
+        status: "error",
+        documents: [],
+        message: `Comment ${commentDocRef.id} does not exist`,
+      };
+    }
+
+    const commentLogicResultDoc: LogicResultDoc = {
+      action: "merge",
+      dstPath: commentDocRef.path,
+      instructions: {
+        repliesCount: "--",
+      },
+    };
+    const {
+      "@id": commentId,
+      createdBy: {"@id": commentAuthorId},
+    } = commentDoc;
+
+    const notificationDocPath = `users/
+    ${commentAuthorId}/notifications/${commentId}`;
+
+    const notificationLogicResultDoc: LogicResultDoc = {
+      action: "delete",
+      dstPath: notificationDocPath,
+    };
+    const replyLogicResultDoc: LogicResultDoc = {
+      action: "delete",
+      dstPath: docPath,
+      doc: replyDoc,
```

Comment:
Hi jeremy just a minor correction. you dont need to have doc field if the action is delete.